embrace-android-api/src/main/kotlin/io/embrace/android/embracesdk/internal/capture/activity/OpenEvents.kt

```diff
@@ -0,0 +1,62 @@
+package io.embrace.android.embracesdk.internal.capture.activity
+
+/**
+ * The relevant stages in the lifecycle of Activities pertaining to observing the performance of their loading
+ */
+public interface OpenEvents {
+
+    /**
+     * When a previously in-progress Activity Open trace should be abandoned, and that the component managing
+     * the trace recording should prepare itself to start tracing the opening of a new Activity instance.
```

Comment:
An activity can be passed an intent and alter its behavior depending on the bundle inside - I don't think it's safe to assume 1 activity type == only 1 instance at a time. I've certainly written apps in the past that host different fragments etc based on intent args, so there may be apps in the wild that made similar decisions

Reply:
Got it. So the case you're concerned about is FooActivity opening another instance of FooActivity?

I /believe/ this case is handled correctly if the first instance completes its lifecycle before starting another instance of the same activity. I will write a test to verify this.

The case I was talking about is regarding a new Activity instance beginning to initialize before the previous one is paused, in which case the unfinished instance, regardless of type, will be assumed to be abandoned. 

For the startup trace, I had initially made the incorrect assumption that if the type is the same that it must be the same instance, which is a subset of the case you're talking about and cannot be assumed. This is why I'm using the hash code rather than the type, to prevent the case where a new implementation of FooActivity is started before the previous instance has completed